Command logging in console option in config

This is a feature request to add a config option to be able to enable/disable command block/player command logging in the console. This gets incredibly annoying when you have a command block clock especially since it is a default spigot.yml feature. Usually I would edit the source code, but I get this:

[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:23 min
[INFO] Finished at: 2015-08-21T11:36:43-03:00
[INFO] Final Memory: 17M/316M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.
3.2:compile (default-compile) on project Essentials: Compilation failure: Compil
ation failure:
[ERROR] \Users_\Documents\GitHub\Essentials\Essentials\src\com\earth2me
\essentials\UserMap.java:[117,15] error: method get in interface Cache<K,V> cann
ot be applied to given types;
[ERROR] \Users_\Documents\GitHub\Essentials\Essentials\src\com\earth2me
\essentials\commands\Commandbreak.java:[22,36] error: reference to getTargetBloc
k is ambiguous
[ERROR] \Users_\Documents\GitHub\Essentials\Essentials\src\com\earth2me
\essentials\craftbukkit\FakeWorld.java:[20,7] error: FakeWorld is not abstract a
nd does not override abstract method getWorldBorder() in World
[ERROR] \Users_\Documents\GitHub\Essentials\Essentials\src\com\earth2me
\essentials\commands\Commandlightning.java:[28,50] error: reference to getTarget
Block is ambiguous
[ERROR] \Users_\Documents\GitHub\Essentials\Essentials\src\com\earth2me
\essentials\OfflinePlayer.java:[27,7] error: OfflinePlayer is not abstract and d
oes not override abstract method getLastTwoTargetBlocks(Set<Material>,int) in Li
vingEntity
[ERROR] \Users_\Documents\GitHub\Essentials\Essentials\src\com\earth2me
\essentials\signs\SignPlayerListener.java:[46,35] error: reference to getTargetB
lock is ambiguous

A config option or help on compiling the source code would really improve things (for me at least) And I know some other people were upset at how the official essentials or Teamcity essentials didnt have this as an option.

On a side note, is Spigot-essentials required / needed with essX or do the features come with it?

Thanks for the help!

--- a/Essentials/src/com/earth2me/essentials/Essentials.java
+++ b/Essentials/src/com/earth2me/essentials/Essentials.java
@@ -487,7 +487,9 @@ public class Essentials extends JavaPlugin implements net.ess3.api.IEssentials {
             }
 
             if (bSenderBlock != null) {
-                Bukkit.getLogger().log(Level.INFO, "CommandBlock at {0},{1},{2} issued server command: /{3} {4}", new Object[]{bSenderBlock.getX(), bSenderBlock.getY(), bSenderBlock.getZ(), commandLabel, EssentialsCommand.getFinalArg(args, 0)});
+                if (getSettings().doCommandBlockCommandLogging()) {
+                    Bukkit.getLogger().log(Level.INFO, "CommandBlock at {0},{1},{2} issued server command: /{3} {4}", new Object[]{bSenderBlock.getX(), bSenderBlock.getY(), bSenderBlock.getZ(), commandLabel, EssentialsCommand.getFinalArg(args, 0)});
+                }
             } else if (user == null) {
                 Bukkit.getLogger().log(Level.INFO, "{0} issued server command: /{1} {2}", new Object[]{cSender.getName(), commandLabel, EssentialsCommand.getFinalArg(args, 0)});
             }

--- a/Essentials/src/com/earth2me/essentials/Essentials.java
+++ b/Essentials/src/com/earth2me/essentials/Essentials.java
@@ -487,7 +487,7 @@ public class Essentials extends JavaPlugin implements net.ess3.api.IEssentials {
             }
 
             if (bSenderBlock != null) {
-                if (getSettings().doCommandBlockCommandLogging()) {
+                if (getSettings().logCommandBlockCommands()) {
                     Bukkit.getLogger().log(Level.INFO, "CommandBlock at {0},{1},{2} issued server command: /{3} {4}", new Object[]{bSenderBlock.getX(), bSenderBlock.getY(), bSenderBlock.getZ(), commandLabel, EssentialsCommand.getFinalArg(args, 0)});
                 }
             } else if (user == null) {

--- a/Essentials/src/com/earth2me/essentials/ISettings.java
+++ b/Essentials/src/com/earth2me/essentials/ISettings.java
@@ -329,5 +329,5 @@ public interface ISettings extends IConf {
 
     boolean isSafeUsermap();
 
-    boolean doCommandBlockCommandLogging();
+    boolean logCommandBlockCommands();
 }

--- a/Essentials/src/com/earth2me/essentials/ISettings.java
+++ b/Essentials/src/com/earth2me/essentials/ISettings.java
@@ -328,4 +328,6 @@ public interface ISettings extends IConf {
     boolean allowOldIdSigns();
 
     boolean isSafeUsermap();
+
+    boolean doCommandBlockCommandLogging();
 }

--- a/Essentials/src/com/earth2me/essentials/Settings.java
+++ b/Essentials/src/com/earth2me/essentials/Settings.java
@@ -544,7 +544,7 @@ public class Settings implements net.ess3.api.ISettings {
         forceEnableRecipe = _isForceEnableRecipe();
         allowOldIdSigns = _allowOldIdSigns();
         isSafeUsermap = _isSafeUsermap();
-        doCommandBlockCommandLogging = _doCommandBlockCommandLogging();
+        logCommandBlockCommands = _logCommandBlockCommands();
     }
 
     void _lateLoadItemSpawnBlacklist() {
@@ -1554,14 +1554,14 @@ public class Settings implements net.ess3.api.ISettings {
         return isSafeUsermap;
     }
 
-    private boolean doCommandBlockCommandLogging;
+    private boolean logCommandBlockCommands;
 
-    private boolean _doCommandBlockCommandLogging() {
+    private boolean _logCommandBlockCommands() {
         return config.getBoolean("log-command-block-commands", true);
     }
 
     @Override
-    public boolean doCommandBlockCommandLogging() {
-        return doCommandBlockCommandLogging;
+    public boolean logCommandBlockCommands() {
+        return logCommandBlockCommands;
     }
 }

--- a/Essentials/src/com/earth2me/essentials/Settings.java
+++ b/Essentials/src/com/earth2me/essentials/Settings.java
@@ -544,6 +544,7 @@ public class Settings implements net.ess3.api.ISettings {
         forceEnableRecipe = _isForceEnableRecipe();
         allowOldIdSigns = _allowOldIdSigns();
         isSafeUsermap = _isSafeUsermap();
+        doCommandBlockCommandLogging = _doCommandBlockCommandLogging();
     }
 
     void _lateLoadItemSpawnBlacklist() {
@@ -1551,5 +1552,16 @@ public class Settings implements net.ess3.api.ISettings {
     @Override
     public boolean isSafeUsermap() {
         return isSafeUsermap;
+    }
+
+    private boolean doCommandBlockCommandLogging;
+
+    private boolean _doCommandBlockCommandLogging() {
+        return config.getBoolean("log-command-block-commands", true);
+    }
+
+    @Override
+    public boolean doCommandBlockCommandLogging() {
+        return doCommandBlockCommandLogging;
     }
 }

--- a/Essentials/src/config.yml
+++ b/Essentials/src/config.yml
@@ -563,6 +563,10 @@ allow-world-in-broadcastworld: true
 # You should only change this to false if you use Minecraft China.
 safe-usermap-names: true
 
+# Should Essentials output logs when a command block executes a command?
+# Example: CommandBlock at <x>,<y>,<z> issued server command: /<command>
+log-command-block-commands: true
+
 ############################################################
 # +------------------------------------------------------+ #
 # |                   EssentialsHome                     | #

